### ISSUE TITLE
EpisodeFileDeletedEvent errors filling up the logs

### ISSUE THREAD

**Issue by breakthestatic** [Sonarr/Sonarr#4289]
**Describe the bug**

My log is being filled with errors like the following for several of my series:
```
21-1-28 12:22:48.1|Error|EventAggregator|NotificationService failed while processing [EpisodeFileDeletedEvent]

[v3.0.4.1091] System.InvalidOperationException: Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00010] in <798dd06330af4ca3886d7d2be5573af7>:0 
  at NzbDrone.Core.Notifications.NotificationService.GetMessage (NzbDrone.Core.Tv.Series series, System.Collections.Generic.List`1[T] episodes, NzbDrone.Core.Qualities.QualityModel quality) [0x0010c] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Notifications\NotificationService.cs:71 
  at NzbDrone.Core.Notifications.NotificationService.Handle (NzbDrone.Core.MediaFiles.Events.EpisodeFileDeletedEvent message) [0x00006] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Notifications\NotificationService.cs:199 
  at NzbDrone.Core.Messaging.Events.EventAggregator.PublishEvent[TEvent] (TEvent event) [0x0011a] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Messaging\Events\EventAggregator.cs:92 
```

**To Reproduce**

Occurs on its own several times a day, but can be manually invoked by clicking the "Update All" button on the series list landing page.

**Expected behavior**

No errors to occur

**Platform Information (please complete the following information):**
 - OS: Ubuntu 20.04
 - Docker: Yes
 - Mono 5.20.1.34
 - Sonarr Version: 3.0.4.1091
 - Sonarr Branch: phantom-develop

**Trace Logs**
[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/5889866/sonarr.trace.txt)

**Comment by bakerboy448** [comment]
@RobinDadswell @markus101 if it is in fact a a bug seems related to the recent delete additions?

**Comment by RobinDadswell** [comment]
> @RobinDadswell @markus101 if it is in fact a a bug seems related to the recent delete additions?

@bakerboy448  I am not getting it on mine so don't believe so as I have had a few things delete today as part of an upgrade

Looking at the trace log I am seeing `21-1-28 12:22:48.0|Debug|MediaFileTableCleanupService|File [/storage/tv/Modern Family/Season 10/Modern.Family.S10E01.720p.HDTV.x264-KILLERS.mkv] is not assigned to any episodes, removing from db` so there is definitely an episode being deleted from the database but not the file system which is triggering this.

**Comment by cdonovan** [comment]
I don't know if this is a red herring, but if it provides any more insight - the particular series was imported from an existing folder, and had a mix of 1080p & 720p files. I had set the monitoring quality to 1080p.  I see that all the errors in the log are for the 720p files only.

Might just be coincidence, but I wanted to share the info in the even it helps debug further.

**Comment by breakthestatic** [comment]
I don't know if this is a red herring, but if it provides any more insight - the particular series was imported from an existing folder, and had a mix of 1080p & 720p files. I had set the monitoring quality to 1080p. I see that all the errors in the log are for the 720p files only.

Might just be coincidence, but I wanted to share the info in the even it helps debug further.

**Comment by c1p0** [comment]
I can confirm I am having the same error coming up a bunch of times a day and it can be invoked by clicking "Update All"

- Ubuntu 20.04
- Docker: No
- Version 3.0.4.1095
- Package Version 3.0.3
- Mono Version 6.10.0.104
- Branch: phantom-develop (i think)

I've started looking at the log files because for some time there are a few series for which episodes don't download automatically even if everything matches. I need to click the icon for the download to begin. Not sure if this has anything to do with it.

**Comment by breakthestatic** [comment]
@bakerboy448, @RobinDadswell or @markus101, it seems that (at least in my case) the MediaFileTableCleanupService errors are occurring when I have duplicate episodes in the folder.  I'm not sure how or when this occurred, but it seems that one file was the original downloaded name, and the other was the Sonarr-renamed file, e.g.

Season 10:
  Modern.Family.S10E01.720p.HDTV.x264-KILLERS.mkv
  Modern Family - S10E01 720p HDTV.mkv

I tried removing one of the duplicate files and running the update scan again, but it seems that the db hangs on to the old references (even if I delete the series and re-add it).  The only way I was able to get rid of the errors was by starting with a completely fresh Sonarr install.
